fireant/tests/test_set_helpers.py

```diff
@@ -0,0 +1,380 @@
+from unittest import TestCase
+
+import pandas as pd
+from pypika import Table
+
+from fireant import (
+    DataSet,
+    DataType,
+    Field,
+    Set,
+)
+from fireant.set_helpers import apply_set_filters
+from fireant.tests.database.mock_database import TestDatabase
+
+
+class SetFilterTests(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        db = TestDatabase()
+        t0 = Table("test0")
+        cls.dataset = DataSet(
+            table=t0,
+            database=db,
+            fields=[
+                Field(
+                    "timestamp",
+                    label="Timestamp",
+                    definition=t0.timestamp,
+                    data_type=DataType.date,
+                ),
+                Field(
+                    "is_metric",
+                    label="Is Metric",
+                    definition=t0.is_metric,
+                    data_type=DataType.boolean,
+                ),
+                Field(
+                    "metric_desc",
+                    label="Metric Description",
+                    definition=t0.metric_desc,
+                    data_type=DataType.text,
+                ),
+                Field(
+                    "metric0",
+                    label="Metric0",
+                    definition=t0.metric,
+                    data_type=DataType.number,
+                ),
+            ],
+        )
+
+    def setUp(self):
+        self.df_dict = {
+            "$metric0": [1, 2, 3, 4],
+            "$is_metric": [True, False, False, False],
+            "$metric_desc": ["text1", "text2", "text3", "text4"],
+        }
+        self.df = pd.DataFrame.from_dict(self.df_dict)
+
+    def test_set_filter_with_greater_than(self):
+        set_filter = Set(self.dataset.fields.metric0 > 2)
+
+        result_df = apply_set_filters(self.df, set_filter)
+
+        pd.testing.assert_frame_equal(
+            result_df.reset_index(drop=True),
+            pd.DataFrame.from_dict(
+                {
+                    **self.df_dict,
+                    "$set(metric>2)": [
+                        "out_set(metric>2)",
```

Comment:
I can add an extra param to the Set constructor, so that in and out values can be provided. Or perhaps we can just assume null in one of them, whenever the other is specified. If both are specified that's fine too. Anyway, if they specify a name, I don't think we need the in_set/out_set or similar prefix.

Reply:
If a parameter was provided to the set e.g. 'top_destinations' as an example, we'd still need to display which are in that set and which are not though? e.g. In top destinations, outside top destinations or something like that.

Reply:
So I imagine 3 scenarios here. When they name both in and out, in which case we just use their provided name. If they just name one (in or our). We can simply use null for the other as it is probably not very important. Like in top_destinations. So they'd see the top destinations with top_destinations written and just empty string for the rest.

Reply:
Actually scratch that, there are use cases where we'd only want to see just a series for the grouped items and no 'out' series for all of the rest. So a null could exclude it.